docs/how-to/hip_porting_guide.rst

```diff
@@ -0,0 +1,715 @@
+etc., needed to compile a HIP program, this includes host as well as device
+code. ``hip_runtime_api.h`` is a subset of ``hip_runtime.h``.
+
+CUDA has slightly different contents for these two files. In some cases you may
+need to convert hipified code to include the richer ``hip_runtime.h`` instead of
+``hip_runtime_api.h``.
+
+Using a Standard C++ Compiler
+--------------------------------------------------------------------------------
+
+You can compile ``hip_runtime_api.h`` using a standard C or C++ compiler
+(e.g., ``gcc`` or ``icc``).
+A source file that is only calling HIP APIs but neither defines nor launches any
+kernels can be compiled with a standard host compiler (e.g. ``gcc`` or ``icc``)
+even when ``hip_runtime_api.h`` or ``hip_runtime.h`` are included.
+
+The HIP include paths and platform macros (``__HIP_PLATFORM_AMD__`` or
+``__HIP_PLATFORM_NVIDIA__``) must be passed to the compiler.
+
+``hipconfig`` can help in finding the necessary options, for example on an AMD
+platform:
+
+.. code-block:: bash
+
+  hipconfig --cpp_config
+   -D__HIP_PLATFORM_AMD__= -I/opt/rocm/include
+
+``nvcc`` includes some headers by default. ``hipcc`` does not include
+default headers, and instead all required files must be explicitly included.
+
+The ``hipify`` tool automatically converts ``cuda_runtime.h`` to
+``hip_runtime.h``, and it converts ``cuda_runtime_api.h`` to
+``hip_runtime_api.h``, but it may miss nested headers or macros.
+
+warpSize
+================================================================================
+
+Code should not assume a warp size of 32 or 64, as that is not portable between
+platforms and architectures. The ``warpSize`` built-in should be used in device
+code, while the host can query it during runtime via the device properties. See
+the :ref:`HIP language extension for warpSize <warp_size>` for information on
+how to write portable wave-aware code.
+
+Lane masks bit-shift
+================================================================================
+
+A thread in a warp is also called a lane, and a lane mask is a bitmask where
+each bit corresponds to a thread in a warp. A bit is 1 if the thread is active,
+0 if it's inactive. Bit-shift operations are typically used to create lane masks
+and on AMD GPUs the ``warpSize`` can differ between different architectures,
+that's why it's essential to use correct bitmask type, when porting code.
+
+Example:
+
+.. code-block:: cpp
+
+  // Get the thread's position in the warp
+  unsigned int laneId = threadIdx.x % warpSize;
+
+  // Use lane ID for bit-shift
+  val & ((2 << (threadIdx.x % warpSize) )-1 );
+
+  // Shift 32 bit integer with val variable
+  WarpReduce::sum( (val < warpSize) ? (1 << val) : 0);
+
+Lane masks are 32-bit integer types as this is the integer precision that C 
+assigns to such constants by default. GCN/CDNA architectures have a warp size of
+64, :code:`threadIdx.x % warpSize` and :code:`val` in the example may obtain 
```

Comment:
GCN/CDNA architectures have a warpsize of 64 by default?

Reply:
GCN/CDNA architectures have a warpsize of 64, that's not changeable at GCN/CDNA. So no need for the by default.